include/boost/geometry/io/wkt/write.hpp

```diff
@@ -62,22 +63,67 @@ namespace boost { namespace geometry
 namespace detail { namespace wkt
 {
 
+template <typename Output>
+struct output_formatter
+{
+    template <typename T>
+    static void append(Output& out, T const& v)
+    {
+        out << v;
+    }
+};
+
+template <typename CharT, typename Traits, typename Allocator>
+struct output_formatter<std::basic_string<CharT, Traits, Allocator>>
+{
+    template <typename T>
+    static void append(std::basic_string<CharT, Traits, Allocator>& out, T const& val)
+    {
+        std::basic_stringstream<CharT, Traits, Allocator> ss;
+        ss << val;
+        out += ss.str();
+    }
+
+    static void append(std::basic_string<CharT, Traits, Allocator>& out, CharT c)
+    {
+        out += c;
+    }
+
+    static void append(std::basic_string<CharT, Traits, Allocator>& out, const CharT* s)
+    {
+        out += s;
+    }
+
+    template <typename AppendCharT, typename AppendTraits, typename AppendAllocator>
+    static void append(
+        std::basic_string<CharT, Traits, Allocator>& out, 
+        std::basic_string<AppendCharT, AppendTraits, AppendAllocator> const& str
+        )
+    {
+        out += str;
+    }
+};
+
 template <typename P, int I, int Count>
 struct stream_coordinate
 {
-    template <typename Char, typename Traits>
-    static inline void apply(std::basic_ostream<Char, Traits>& os, P const& p)
+    template <typename Output>
+    static inline void apply(Output& out, P const& p)
     {
-        os << (I > 0 ? " " : "") << get<I>(p);
-        stream_coordinate<P, I + 1, Count>::apply(os, p);
+        if (I > 0)
+        {
+            output_formatter<Output>::append(out, ' ');
```

Comment:
So e.g. here `' '` is of type `char` so if the `Output` is `std::basic_string<wchar_t>` then here the `append()` using `std::basic_stringstream` will be instantiated defeating the purpose of having multiple `append()` overloads.

Reply:
Yes, as I also agreed in details in  https://github.com/boostorg/geometry/pull/670#discussion_r387284769